File: python/find_duplicates.py
def find_duplicates_nested_loop(l: list) -> list:
    newList = []
    listSize = len(l)

    for x in range(0, listSize -1):
        for y in range(x+1, listSize):
            if l[x] == l [y]:
                if l[x] not in newList:
                    newList.append(l[x])  # or index = 0
                                          # newlist[index] = l[x]
                                          # index += 1
    
    return newList

if __name__ == "__main__":
    sample1 = [3, 7, 5, 6, 7, 4, 8, 5, 7, 66]
    sample2 = [3, 5, 6, 4, 4, 5, 66, 6, 7, 6]
    sample3 = [3, 0, 5, 1, 0]
    sample4 = [3]
    
    print("Sample 1:", find_duplicates_nested_loop(sample1))
    print("Sample 2:", find_duplicates_nested_loop(sample2))
    print("Sample 3:", find_duplicates_nested_loop(sample3))
    print("Sample 4:", find_duplicates_nested_loop(sample4))

File: java/FindDuplicates.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class FindDuplicates {

	public static List<Integer> findDuplicatesNestedLoops(List<Integer> l) {
		List<Integer> newList = new ArrayList<Integer>();
		int listLength = l.size();
		int addNum;
		
		for(int i = 0; i < listLength - 1; i++) {
			for(int j = i + 1; j < listLength; j++) {
				if(l.get(i).equals(l.get(j))) {
					addNum = l.get(j);
					if (!newList.contains(addNum)) {
						newList.add(addNum);
					}
				}
			}
		}
		return newList;
	}

	public static void main(String[] args) {
		// some test strings:
		List<Integer> sample1 = new ArrayList<Integer>(Arrays.asList(3, 7, 5, 6, 7, 4, 8, 5, 7, 66));
		List<Integer> sample2 = new ArrayList<Integer>(Arrays.asList(3, 5, 6, 4, 4, 5, 66, 6, 7, 6));
		List<Integer> sample3 = new ArrayList<Integer>(Arrays.asList(3, 0, 5, 1, 0));
		List<Integer> sample4 = new ArrayList<Integer>(Arrays.asList(3));
		System.out.println("Sample 1: " + findDuplicatesNestedLoops(sample1));
		System.out.println("Sample 2: " + findDuplicatesNestedLoops(sample2));
		System.out.println("Sample 3: " + findDuplicatesNestedLoops(sample3));
		System.out.println("Sample 4: " + findDuplicatesNestedLoops(sample4));
	}

}